papers/yrcs-talk-content.tex

```diff
@@ -0,0 +1,300 @@
+	\item Stand-alone key exchange app
+	\item A Sci-Comm project teaching people about post-quantum security
+\end{itemize}
+\end{frame}
+
+\begin{frame}{WireGuard\footnote{Jason A. Donenfeld. “WireGuard: Next Generation Kernel Network Tunnel”. In: 24th Annual Network and Distributed System Security Symposium, NDSS 2017, San Diego, California, USA, February 26 - March 1, 2017. Whitepaper: https: //www.wireguard.com/papers/wireguard.pdf.}}
+\begin{itemize}
+  \item VPN protocol in the linux kernel
+  \item Based on Noise IKpsk1 from the Noise Protocol Framework\footnote{Trevor Perrin. The Noise Protocol Framework. 2016. url: http://noiseprotocol.org/noise.pdf}
+  \item Small, fast, open source crypto
+\end{itemize}
+\end{frame}
+
+\begin{frame}{WireGuard/Noise IKpsk security properties}
+\begin{itemize}
+  \itemtick Session-key secrecy
+  \itemtick Forward-secrecy
+  \itemtick Mutual authentication
+  \itemtick Session-key Uniqueness
+  \itemtick Identity Hiding
+  \itemtick (DoS Mitigation – First packet is authenticated\footnote{Based on the unrealistic assumption of a monotonic counter – We found a practical attack})
+\end{itemize}
+\end{frame}
+
+\begin{frame}{Security of Rosenpass}
+\begin{columns}
+
+\begin{column}{.30\textwidth}
+WireGuard
+\begin{itemize}
+  \itemtick Session-key secrecy
+  \itemtick Forward-secrecy
+  \itemtick Mutual authentication
+  \itemtick Session-key Uniqueness
+  \itemtick Identity Hiding
+  \itemtick (DoS Mitigation)
+\end{itemize}
+\end{column}
+
+\begin{column}{.30\textwidth}
+Post-Quantum WireGuard
+\begin{itemize}
+  \itemfail Identity Hiding \footnote{Based on a Identity Hiding/ANON-CCA security of McEliece; unclear whether that holds.}
+  \itemfail DoS Mitigation \footnote{PQWG provides DoS mitigation under the assumption of a secret PSK, which quite frankly is cheating.}
+\end{itemize}
+\end{column}
+
+\begin{column}{.30\textwidth}
+Rosenpass
+\begin{itemize}
+  \itemtick DoS Mitigation
+  \itemtick Hybrid Post-Quantum security\footnote{In deployments using WireGuard + Rosenpass; Rosenpass on its own provides post-quantum security.}
+\end{itemize}
+\end{column}
+
+\end{columns}
+\end{frame}
+
+\begin{frame}{Building post-quantum WireGuard: NIKE vs KEM}
+
+
+  NIKE:
+
+$(\texttt{sk}_1, \texttt{pk}_1) \leftarrow \texttt{NIKE.KeyGen}$ \\
+$(\texttt{sk}_2, \texttt{pk}_2) \leftarrow \texttt{NIKE.KeyGen}$ \\
+$\texttt{NIKE.SharedKey}(\texttt{sk}_1, \texttt{pk}_2) = \texttt{NIKE.SharedKey}(\texttt{sk}_2, \texttt{pk}_1)$
+
+  KEM:
+
+$(\texttt{sk}, \texttt{pk}) \leftarrow \texttt{KEM.KeyGen}$ \\
+$(\texttt{shk}, \texttt{ct}) \leftarrow \texttt{KEM.Encaps}(\texttt{pk})$ \\
+$\texttt{shk} = \texttt{KEM.Decaps}(\texttt{sk}, \texttt{ct})$
+
+\end{frame}
+
+
+
+\begin{frame}{Minimal key exchange using KEMs}
+\begin{tikzpicture}[shorten > = 1pt,shorten < = 1pt]
+	\draw (-3,0) node[above](initiator){Initiator\strut} -- coordinate[pos=.2](pk-y) coordinate[pos=.6](ct-y)coordinate[pos=.8](ack-y)+(0,-5);
```

Comment:
I changed those to relative placements instead of absolute ones. We will probably need to reuse the structuture,